Source/WebCore/platform/MediaSample.h

```diff
@@ -206,26 +204,46 @@ struct AudioInfo : public TrackInfo {
     bool equalTo(const TrackInfo& otherAudioInfo) const final
     {
         auto& other = downcast<const AudioInfo>(otherAudioInfo);
-        return rate == other.rate && channels == other.channels && bitDepth == other.bitDepth && framesPerPacket == other.framesPerPacket && profile == other.profile && extendedProfile == other.extendedProfile && ((!cookieData && !other.cookieData) || (cookieData && other.cookieData && *cookieData == *other.cookieData));
+        return rate == other.rate && channels == other.channels && bitDepth == other.bitDepth && framesPerPacket == other.framesPerPacket && ((!cookieData && !other.cookieData) || (cookieData && other.cookieData && *cookieData == *other.cookieData));
     }
 };
 
 class MediaSamplesBlock {
+    WTF_MAKE_FAST_ALLOCATED;
 public:
-    using MediaSampleDataType = std::variant<MediaSample::ByteRange, Ref<const FragmentedSharedBuffer>>;
+    using MediaSampleDataType = RefPtr<FragmentedSharedBuffer>;
     struct MediaSampleItem {
         using MediaSampleDataType = MediaSamplesBlock::MediaSampleDataType;
         MediaTime presentationTime;
-        MediaTime decodeTime;
-        MediaTime duration;
-        MediaTime trimDuration;
+        MediaTime decodeTime { MediaTime::indefiniteTime() };
+        MediaTime duration { MediaTime::zeroTime() };
+        std::pair<MediaTime, MediaTime> trimInterval { MediaTime::zeroTime(), MediaTime::zeroTime() };
         MediaSampleDataType data;
-        MediaSample::SampleFlags flags;
+        uint32_t flags { };
+        bool isSync() const { return flags & MediaSample::IsSync; }
     };
     using SamplesVector = Vector<MediaSampleItem>;
 
+    MediaSamplesBlock() = default;
+    MediaSamplesBlock(const TrackInfo* info, SamplesVector&& items)
+        : m_info(info)
+        , m_samples(WTFMove(items))
+    {
+    }
+
     void setInfo(RefPtr<const TrackInfo>&& info) { m_info = WTFMove(info); }
     const TrackInfo* info() const { return m_info.get(); }
+    MediaTime presentationTime() const { return isEmpty() ? MediaTime::invalidTime() : first().presentationTime; }
+    MediaTime duration() const
+    {
+        MediaTime duration = MediaTime::zeroTime();
+        for (auto& sample : *this)
+            duration += sample.duration;
+        return duration;
+    }
+    MediaTime presentationEndTime() const { return presentationTime() + duration(); }
+    bool isSync() const { return size() ? (first().flags & MediaSample::IsSync) : false; }
+    TrackID trackID() const { return m_info ? m_info->trackID : -1; }
```

Comment:
Might be worth to add a constant for -1, or use a std::optional<TrackID> instead.